sdk/servicebus/Azure.Messaging.ServiceBus/tests/Receiver/ReceiverTests.cs

```diff
@@ -121,6 +121,22 @@ public void ReceiveValidatesMaxWaitTime()
                 Throws.InstanceOf<ArgumentOutOfRangeException>());
         }
 
+        [Test]
+        public void ReceiveValidatesMaxWaitTimePrefetchMode()
+        {
+            var account = Encoding.Default.GetString(GetRandomBuffer(12));
+            var fullyQualifiedNamespace = new UriBuilder($"{account}.servicebus.windows.net/").Host;
+            var connString = $"Endpoint=sb://{fullyQualifiedNamespace};SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey={Encoding.Default.GetString(GetRandomBuffer(64))}";
+            var client = new ServiceBusClient(connString);
+            var receiver = client.CreateReceiver("queue", options: new ServiceBusReceiverOptions { PrefetchCount = 10 });
+            Assert.That(
+                async () => await receiver.ReceiveMessageAsync(TimeSpan.FromSeconds(0)),
```

Comment:
If we're unable to mock out the transport receiver, this is probably better done as a live test.   I'm not overly comfortable with this approach.